artifact-size-enforcer/src/main/java/org/codice/ddf/admin/beta/enforcer/ArtifactSizeEnforcerRule.java

```diff
@@ -0,0 +1,202 @@
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.beta.enforcer;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+
+import org.apache.maven.enforcer.rule.api.EnforcerRule;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleException;
+import org.apache.maven.enforcer.rule.api.EnforcerRuleHelper;
+import org.apache.maven.plugin.logging.Log;
+import org.codehaus.plexus.component.configurator.expression.ExpressionEvaluationException;
+import org.codehaus.plexus.util.StringUtils;
+
+/**
+ * When the ArtifactSizeEnforceRule is active, it will look up the packaging type and enforce a maximum artifact size.
+ * If the packing type is unknown to the rule, enforcement will be skipped. The only currently supported packing types are bundle and jar.
+ *
+ * The following arguments can be specified to the ArtifactSizeEnforcerRule:
+ *
+ * maxArtifactSize - If the specified artifact is larger than this maxArtifactSize then an exception will be thrown. The unit of this argument should end with `_B` (Bytes), `_KB` (Kilobytes), or `_MB` (MegaBytes). By default, maxArtifactSize is set to 1_MB.
+ * artifactLocation - A path to the file to be checked. If a artifactLocation is not looked up, a default artifact will attempt to be found using the project version, packaging and artifactId.
+ * skip - The ArtifactSizeEnforceRule will not run.
+ */
+public class ArtifactSizeEnforcerRule implements EnforcerRule {
+
+    public static final String PROJECT_PACKAGING_PROP = "${project.packaging}";
+    public static final String PROJECT_ARTIFACT_ID_PROP = "${project.artifactId}";
+    public static final String PROJECT_VERSION_PROP = "${project.version}";
+    public static final String PROJECT_BUILD_DIR_PROP = "${project.build.directory}";
+
+    public static final String DEFAULT_MAX_ARTIFACT_SIZE = "1_MB";
+    public static final String BYTES = "_B";
+    public static final String MEGA_BYTES = "_MB";
+    public static final String KILO_BYTES = "_KB";
+
+    public static final String JAR = "jar";
+    public static final String BUNDLE = "bundle";
+    public static final List<String> SUPPORTED_PACKAGE_TYPES = Arrays.asList(JAR, BUNDLE);
+
+    //Rule arguments
+    private String maxArtifactSize;
+    private String artifactLocation;
+    private boolean skip;
+
+    private EnforcerRuleHelper helper;
```

Comment:
❗️ Not thread-safe; won't work correctly in a parallel build.

Same applies for your `logger`.

Since you only receive the `helper` (and the `logger` through that) through one route (the public parameter to the `execute` method, you can just pass them along to the private methods instead of using fields.

Conversely, you could store them in `ThreadLocal`s, but that's an awful lot of effort for one or two params to pass to privates.